react/training/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,100 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 
 import Board from './components/Board';
+import Button from './components/Button';
 import style from './styles.scss';
 
-export default class Game extends Component {
+class Game extends Component {
+  state = {
+    history: [{ squares: Array(9).fill(null) }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  getConst = () => {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = this.calculateWinner(current.squares);
+    const moves = history.map((step, move) => {
+      const desc = move ? `Go to move #${move}` : 'Go to game start';
+      return (
+        <Button
+          key={`autoKey-${move + 1}`}
+          className={style.buttonHistory}
+          clickParam={move}
+          onClick={this.jumpTo}
+        >
+          {desc}
+        </Button>
+      );
+    });
+    return { current, winner, moves };
+  };
+
+  getStatus = squares => {
+    const winner = this.calculateWinner(squares);
+    let status;
+    if (winner) {
+      status = `Winner: ${winner.player}`;
+    } else if (this.state.stepNumber === squares.length) {
+      status = `Nobody wins`;
+    } else {
+      status = `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
+    }
+    return status;
+  };
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (this.calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{ squares }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    });
+  };
+
+  calculateWinner = squares => {
+    const lines = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
```

Comment:
I forgot to comment on this in the previous pull request. Extract this `lines` to a constant file in `src/constants` folder. This lines never change and it ins't necessary declare it in each function call